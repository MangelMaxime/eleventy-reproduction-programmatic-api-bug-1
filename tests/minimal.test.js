import { expect, test } from 'vitest';
import Eleventy from "@11ty/eleventy";

test('minimal', async () => {
    console.log(process.cwd());
    const elev = new Eleventy(
        "./tests/fixtures/minimal/",
        "./tests/fixtures/minimal/_site",
        {
            configPath: "./tests/fixtures/minimal/.eleventy.js"
        } 
    );

    const json = await elev.toJSON(); 

    expect(json).toMatchSnapshot();
})
 