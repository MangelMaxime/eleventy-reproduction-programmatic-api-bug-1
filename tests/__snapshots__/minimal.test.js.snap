// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`minimal 1`] = `
[
  {
    "content": "<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Hello</title>
</head>
<body>
    <div class="layout" role="document">
        <main id="main" class="main">
            <h1>Hello</h1>

        </main>
    </div>
</body>
</html>
",
    "inputPath": "/Users/mmangel/Workspaces/Github/MangelMaxime/eleventy-reproduction-programmatic-api-bug-1/tests/fixtures/minimal/_site/index.html",
    "outputPath": "./_site/_site/index.html",
    "rawInput": "<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Hello</title>
</head>
<body>
    <div class="layout" role="document">
        <main id="main" class="main">
            <h1>Hello</h1>

        </main>
    </div>
</body>
</html>
",
    "url": "/_site/",
  },
  {
    "content": "<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Hello</title>
</head>
<body>
    <div class="layout" role="document">
        <main id="main" class="main">
            <h1>Index page</h1>

        </main>
    </div>
</body>
</html>
",
    "inputPath": "/Users/mmangel/Workspaces/Github/MangelMaxime/eleventy-reproduction-programmatic-api-bug-1/tests/fixtures/minimal/index.md",
    "outputPath": "./_site/index.html",
    "rawInput": "
# Index page
",
    "url": "/",
  },
]
`;
